src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -193,10 +193,26 @@ void TelemetryImpl::enable()
     //        For now, we just do the same as QGC does.
 
     _parent->add_call_every([this]() { check_calibration(); }, 5.0, &_calibration_cookie);
+
+    // We're going to retry until we have the Home Position.
+    _parent->add_call_every([this]() { request_home_position_again(); }, 2.0f, &_homepos_cookie);
 }
 
 void TelemetryImpl::disable() {}
 
+void TelemetryImpl::request_home_position_again()
+{
+    {
+        std::lock_guard<std::mutex> lock(_request_home_position_mutex);
+        if (_health.is_home_position_ok) {
+            _parent->remove_call_every(_homepos_cookie);
```

Comment:
Don't forget to also call this `_parent->remove_call_every(_homepos_cookie);` in `disable()`, and maybe add the one for the calibration as well, thanks!

Reply:
Thanks :) just added to the latest commit.